publish_trips.py

```diff
@@ -151,6 +152,18 @@ def populate_month_year(data):
         row["month"] = date.strftime("%-m")
     return data
 
+def datetime_creation(data):
+    """
+    Joins together the checkout time and date columns to a single datetime field.
+    """
+    for row in data:
+        # Hour is not zero padded in the data so we have to check for that
+        if len(row["checkout_time"]) == 7:
+            row["checkout_datetime"] = f"{row['checkout_date']}T0{row['checkout_time']}"
+        else:
+            row["checkout_datetime"] = f"{row['checkout_date']}T{row['checkout_time']}"
+    return data
```

Comment:
just thinking about data validation, if this datetime string ends up being an invalidate date, Socrata will hopefully throw an error, right?

what you have is great but `zfill` is a fun string method:
```python
>>> "1:30:00".zfill(8)
'01:30:00'
```

Reply:
Ah TIL about `zfill`!

Socrata will not error out based on my experience if the datetime is an invalid date. I think it'll usually just end up being nulled 👎

Reply:
Now that I think about about some more, we could potentially try to use the Socrata [dataset alerts](https://support.socrata.com/hc/en-us/articles/360000235507-Create-and-Manage-Dataset-Alerts-on-the-Enterprise-Data-Platform) to try to catch these bugs as they enter the dataset in addition to cerberus in the ETL. Should hopefully let us stay on top of these rather than letting them build dust potentially?

Reply:
@roseeichelmann yeah you can query the dataset and if it returns rows it will send an alert email. It's a pretty neat feature but I don't think it's super necessary here, sometime later prehaps.